### PR TITLE
Fix MockFileInfo.FullName - normalize paths

### PR DESCRIPTION
**Pull request by 304NotModified**
Fixes https://github.com/System-IO-Abstractions/System.IO.Abstractions/issues/425

Feedback highly appreciated

### CODE REVIEW

File: System.IO.Abstractions.TestingHelpers/IMockFileDataAccessor.cs [hunk 1]
```diff
@@ -66,5 +66,6 @@ public interface IMockFileDataAccessor
         PathVerifier PathVerifier { get; }
 
         IFileSystem FileSystem { get; }
+        string FixPath(string path, bool checkCaps = false);
```
Comment (304NotModified) [inline System.IO.Abstractions.TestingHelpers/IMockFileDataAccessor.cs]: Note: Not sure if I should use  checpCaps (currently not), and if not, not sure if I should expose checkCaps to the interface

**Comment by 304NotModified** [comment]
@fgreinacher to be clear, this is ready for review 👼

**Comment by fgreinacher** [comment]
I‘m on vacation for a week. Will take a look when I’m back!

**Comment by 304NotModified** [comment]
Have a good holiday!

File: System.IO.Abstractions.TestingHelpers/MockFileInfo.cs [hunk 2]
```diff
@@ -7,11 +7,14 @@ public class MockFileInfo : FileInfoBase
...
         private readonly IMockFileDataAccessor mockFileSystem;
         private string path;
+        private string originalPath;
 
         public MockFileInfo(IMockFileDataAccessor mockFileSystem, string path) : base(mockFileSystem?.FileSystem)
         {
             this.mockFileSystem = mockFileSystem ?? throw new ArgumentNullException(nameof(mockFileSystem));
-            this.path = path ?? throw new ArgumentNullException(nameof(path));
+            this.originalPath = path ?? throw new ArgumentNullException(nameof(path));
+            this.path = mockFileSystem.FixPath(path);
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers/MockFileInfo.cs]: Can this just call `mockFileSystem.Path.GetFullPath(path)`?

File: System.IO.Abstractions.TestingHelpers/IMockFileDataAccessor.cs (hunk 1 above)
Comment (fgreinacher) [reply System.IO.Abstractions.TestingHelpers/IMockFileDataAccessor.cs]: I'd rather not add this somehow unrelated method here and instead try to delegate expanding paths to `PathBase`/`IPath`, see comment below.

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs [hunk 3]
```diff
@@ -536,6 +536,28 @@ public void MockFileInfo_ToString_ShouldReturnOriginalFilePath(string path)
             Assert.AreEqual(filePath, mockFileInfo.ToString());
         }
 
+
+        /// <summary>
+        /// Normalize, tested with Path.GetFullPath and new FileInfo().FullName;
+        /// </summary>
+        [WindowsOnly(WindowsSpecifics.StrictPathRules)]
+        [TestCase(@"c:\top\..\most\file", @"c:\most\file")]
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs]: Is this really Windows-specific?

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs [hunk 4]
```diff
@@ -536,6 +536,28 @@ public void MockFileInfo_ToString_ShouldReturnOriginalFilePath(string path)
...
+        [TestCase(@"c:\top\..\most\..\dir\file", @"c:\dir\file")]
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs]: Dito

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs [hunk 5]
```diff
@@ -536,6 +536,28 @@ public void MockFileInfo_ToString_ShouldReturnOriginalFilePath(string path)
...
+        [TestCase(@"\file", @"C:\file")]
+        [TestCase(@"c:\top\../..\most\file", @"c:\most\file")]
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs]: Dito

File: System.IO.Abstractions.TestingHelpers/MockFileSystem.cs [hunk 6]
```diff
@@ -64,7 +64,7 @@ public MockFileSystem(IDictionary<string, MockFileData> files, string currentDir
         public IFileSystem FileSystem => this;
         public PathVerifier PathVerifier => pathVerifier;
 
-        private string FixPath(string path, bool checkCaps = false)
+        public string FixPath(string path, bool checkCaps = false)
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers/MockFileSystem.cs]: We should not need this IMO (see above)

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs (hunk 3 above)
Comment (304NotModified) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs]: Well drives are Windows only? Not sure how to setup a non-windows test (duplicate test?) and if it's really needed, those will cover the same lines

File: System.IO.Abstractions.TestingHelpers/IMockFileDataAccessor.cs (hunk 1 above)
Comment (304NotModified) [reply System.IO.Abstractions.TestingHelpers/IMockFileDataAccessor.cs]: yes, I was looking for that! Fixed!

File: System.IO.Abstractions.TestingHelpers/MockFileInfo.cs (hunk 2 above)
Comment (304NotModified) [reply System.IO.Abstractions.TestingHelpers/MockFileInfo.cs]: indeed, fixed!

File: System.IO.Abstractions.TestingHelpers/MockFileSystem.cs (hunk 6 above)
Comment (304NotModified) [reply System.IO.Abstractions.TestingHelpers/MockFileSystem.cs]: fixed

**Comment by 304NotModified** [comment]
Changed are done, expect the unit tests. 

PS I think you can asign me nowadays to the PR, would help me in reminder this was still open 👼

**Comment by 304NotModified** [comment]
I see the XFS helper, tried that one

**Comment by 304NotModified** [comment]
All set then 👍 

PS merge master is not really needed if you would squash (recommend!) this one on merge PR . It's a r build so 100 percent safe.

**Comment by fgreinacher** [comment]
Thanks a lot for your contribution 👍

**Pull request merged**